proto/cosmos/staking/v1beta1/tx.proto

```diff
@@ -29,25 +29,57 @@ service Msg {
+
+  // CancelUnbondingDelegation defines a method for performing canceling the unbonding delegation
+  // and delegate back to previous validator.
+  //
+  // This has been backported from SDK 46 as a desirable safety feature for LSM.
+  // If a liquid staking provider is exploited and the exploiter initiates an undelegation,
+  // having access to CancelUnbondingDelegation allows the liquid staking provider to cancel
+  // the undelegation with a software upgrade and thus avoid loss of user funds
+  rpc CancelUnbondingDelegation(MsgCancelUnbondingDelegation) returns (MsgCancelUnbondingDelegationResponse);
+
+  // TokenizeShares defines a method for tokenizing shares from a validator.
+  rpc TokenizeShares(MsgTokenizeShares) returns (MsgTokenizeSharesResponse);
+
+  // RedeemTokensForShares defines a method for redeeming tokens from a validator for
+  // shares.
+  rpc RedeemTokensForShares(MsgRedeemTokensForShares) returns (MsgRedeemTokensForSharesResponse);
+
+  // TransferTokenizeShareRecord defines a method to transfer ownership of
+  // TokenizeShareRecord
+  rpc TransferTokenizeShareRecord(MsgTransferTokenizeShareRecord) returns (MsgTransferTokenizeShareRecordResponse);
+
+  // DisableTokenizeShares defines a method to prevent the tokenization of an addresses stake
+  rpc DisableTokenizeShares(MsgDisableTokenizeShares) returns (MsgDisableTokenizeSharesResponse);
+
+  // EnableTokenizeShares defines a method to re-enable the tokenization of an addresseses stake
+  // after it has been disabled
+  rpc EnableTokenizeShares(MsgEnableTokenizeShares) returns (MsgEnableTokenizeSharesResponse);
+
+  // ValidatorBond defines a method for performing a validator self-bond
+  rpc ValidatorBond(MsgValidatorBond) returns (MsgValidatorBondResponse);
 }
 
 // MsgCreateValidator defines a SDK message for creating a new validator.
 message MsgCreateValidator {
   option (gogoproto.equal)           = false;
   option (gogoproto.goproto_getters) = false;
 
-  Description     description         = 1 [(gogoproto.nullable) = false];
-  CommissionRates commission          = 2 [(gogoproto.nullable) = false];
-  string          min_self_delegation = 3 [
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
-    (gogoproto.moretags)   = "yaml:\"min_self_delegation\"",
-    (gogoproto.nullable)   = false
-  ];
+  Description              description       = 1 [(gogoproto.nullable) = false];
+  CommissionRates          commission        = 2 [(gogoproto.nullable) = false];
   string                   delegator_address = 4 [(gogoproto.moretags) = "yaml:\"delegator_address\""];
   string                   validator_address = 5 [(gogoproto.moretags) = "yaml:\"validator_address\""];
   google.protobuf.Any      pubkey            = 6 [(cosmos_proto.accepts_interface) = "cosmos.crypto.PubKey"];
   cosmos.base.v1beta1.Coin value             = 7 [(gogoproto.nullable) = false];
+  reserved 3;
```

Comment:
Even with this field reserved, I am getting error upon legacy gentx unmarshalling:

```
Error: failed to get genesis app state from config: failed to decode gentx: ... error: can't unmarshal Any nested proto *types.MsgCreateValidator: unknown field "min_self_delegation" in types.MsgCreateValidator: tx parse error
```

To test, simply generate state on v45 without LSM and then try to start with binary with LSM. I am not sure if this is critical or not 🤷‍♂️

Reply:
Let me know if [this](https://github.com/cosmos/cosmos-sdk/pull/16747/commits/df602362b011695941e1aac78bab247ea2c402ed) solves it for you!